### PR TITLE
[Feat] Enabled doctest for CI

### PR DESCRIPTION
**Pull request by shijianjian**
### Description
@edgarriba I fixed most broken doctest but there are some remaining broken ones.

The following two are considered as bugs:
```
/git/kornia/kornia/feature/responses.py:229: DocTestFailure
__________________________________________________________________________________________________ [doctest] kornia.geometry.camera.pinhole.homography_i_H_ref ___________________________________________________________________________________________________
541         pinhole_ref to pinhole_i.
542 
543     Shape:
544         - Input: :math:`(N, 12)` and :math:`(N, 12)`
545         - Output: :math:`(N, 4, 4)`
546 
547     Example:
548         >>> pinhole_i = torch.rand(1, 12)    # Nx12
549         >>> pinhole_ref = torch.rand(1, 12)  # Nx12
550         >>> homography_i_H_ref(pinhole_i, pinhole_ref)  # Nx4x4
UNEXPECTED EXCEPTION: NameError("name 'rtvec_to_pose' is not defined")
Traceback (most recent call last):

  File "/anaconda3/envs/torch/lib/python3.7/doctest.py", line 1329, in __run
    compileflags, 1), test.globs)

  File "<doctest kornia.geometry.camera.pinhole.homography_i_H_ref[2]>", line 1, in <module>

  File "/git/kornia/kornia/geometry/camera/pinhole.py", line 555, in homography_i_H_ref
    i_pose_base = get_optical_pose_base(pinhole_i)

  File "/git/kornia/kornia/geometry/camera/pinhole.py", line 512, in get_optical_pose_base
    return rtvec_to_pose(optical_pose_parent)

NameError: name 'rtvec_to_pose' is not defined

/git/kornia/kornia/geometry/camera/pinhole.py:550: UnexpectedException
___________________________________________________________________________________________ [doctest] kornia.geometry.subpix.spatial_soft_argmax.spatial_soft_argmax2d ___________________________________________________________________________________________
503     Shape:
504         - Input: :math:`(B, N, H, W)`
505         - Output: :math:`(B, N, 2)`
506 
507     Examples:
508         >>> input = torch.tensor([[[
509         ... [0., 0., 0.],
510         ... [0., 10., 0.],
511         ... [0., 0., 0.]]]])
512         >>> spatial_soft_argmax2d(input, False)
Expected:
    tensor([[[1.0000, 1.0000]]])
Got:
    tensor([[[ 0.0000e+00, -1.4901e-08]]])
```

The following three produced different results. Not sure if they were bugged.
```
/git/kornia/kornia/feature/laf.py:292: UnexpectedException
________________________________________________________________________________________________________ [doctest] kornia.feature.responses.gftt_response ________________________________________________________________________________________________________
136         ...    [0., 0., 0., 0., 0., 0., 0.],
137         ...    [0., 1., 1., 1., 1., 1., 0.],
138         ...    [0., 1., 1., 1., 1., 1., 0.],
139         ...    [0., 1., 1., 1., 1., 1., 0.],
140         ...    [0., 1., 1., 1., 1., 1., 0.],
141         ...    [0., 1., 1., 1., 1., 1., 0.],
142         ...    [0., 0., 0., 0., 0., 0., 0.],
143         ... ]]])  # 1x1x7x7
144         >>> # compute the response map
145         >>> gftt_response(input)
Differences (unified diff with -expected +actual):
    @@ -1,7 +1,7 @@
    -tensor([[[[0.0155, 0.0334, 0.0194, 0.0000, 0.0194, 0.0334, 0.0155],
    -          [0.0334, 0.0575, 0.0339, 0.0000, 0.0339, 0.0575, 0.0334],
    -          [0.0194, 0.0339, 0.0497, 0.0000, 0.0497, 0.0339, 0.0194],
    -          [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
    -          [0.0194, 0.0339, 0.0497, 0.0000, 0.0497, 0.0339, 0.0194],
    -          [0.0334, 0.0575, 0.0339, 0.0000, 0.0339, 0.0575, 0.0334],
    -          [0.0155, 0.0334, 0.0194, 0.0000, 0.0194, 0.0334, 0.0155]]]])
    +tensor([[[[0.0379, 0.0456, 0.0283, 0.0121, 0.0283, 0.0456, 0.0379],
    +          [0.0456, 0.0598, 0.0402, 0.0168, 0.0402, 0.0598, 0.0456],
    +          [0.0283, 0.0402, 0.0545, 0.0245, 0.0545, 0.0402, 0.0283],
    +          [0.0121, 0.0168, 0.0245, 0.0276, 0.0245, 0.0168, 0.0121],
    +          [0.0283, 0.0402, 0.0545, 0.0245, 0.0545, 0.0402, 0.0283],
    +          [0.0456, 0.0598, 0.0402, 0.0168, 0.0402, 0.0598, 0.0456],
    +          [0.0379, 0.0456, 0.0283, 0.0121, 0.0283, 0.0456, 0.0379]]]])

/git/kornia/kornia/feature/responses.py:145: DocTestFailure
_______________________________________________________________________________________________________ [doctest] kornia.feature.responses.harris_response _______________________________________________________________________________________________________
050         ...    [0., 0., 0., 0., 0., 0., 0.],
051         ...    [0., 1., 1., 1., 1., 1., 0.],
052         ...    [0., 1., 1., 1., 1., 1., 0.],
053         ...    [0., 1., 1., 1., 1., 1., 0.],
054         ...    [0., 1., 1., 1., 1., 1., 0.],
055         ...    [0., 1., 1., 1., 1., 1., 0.],
056         ...    [0., 0., 0., 0., 0., 0., 0.],
057         ... ]]])  # 1x1x7x7
058         >>> # compute the response map
059         >>> harris_response(input, 0.04)
Differences (unified diff with -expected +actual):
    @@ -1,7 +1,7 @@
    -tensor([[[[0.0012, 0.0039, 0.0020, 0.0000, 0.0020, 0.0039, 0.0012],
    -          [0.0039, 0.0065, 0.0040, 0.0000, 0.0040, 0.0065, 0.0039],
    -          [0.0020, 0.0040, 0.0029, 0.0000, 0.0029, 0.0040, 0.0020],
    -          [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
    -          [0.0020, 0.0040, 0.0029, 0.0000, 0.0029, 0.0040, 0.0020],
    -          [0.0039, 0.0065, 0.0040, 0.0000, 0.0040, 0.0065, 0.0039],
    -          [0.0012, 0.0039, 0.0020, 0.0000, 0.0020, 0.0039, 0.0012]]]])
    +tensor([[[[0.0042, 0.0054, 0.0035, 0.0006, 0.0035, 0.0054, 0.0042],
    +          [0.0054, 0.0068, 0.0046, 0.0014, 0.0046, 0.0068, 0.0054],
    +          [0.0035, 0.0046, 0.0034, 0.0014, 0.0034, 0.0046, 0.0035],
    +          [0.0006, 0.0014, 0.0014, 0.0006, 0.0014, 0.0014, 0.0006],
    +          [0.0035, 0.0046, 0.0034, 0.0014, 0.0034, 0.0046, 0.0035],
    +          [0.0054, 0.0068, 0.0046, 0.0014, 0.0046, 0.0068, 0.0054],
    +          [0.0042, 0.0054, 0.0035, 0.0006, 0.0035, 0.0054, 0.0042]]]])

/git/kornia/kornia/feature/responses.py:59: DocTestFailure
______________________________________________________________________________________________________ [doctest] kornia.feature.responses.hessian_response _______________________________________________________________________________________________________
220         ...    [0., 0., 0., 0., 0., 0., 0.],
221         ...    [0., 1., 1., 1., 1., 1., 0.],
222         ...    [0., 1., 1., 1., 1., 1., 0.],
223         ...    [0., 1., 1., 1., 1., 1., 0.],
224         ...    [0., 1., 1., 1., 1., 1., 0.],
225         ...    [0., 1., 1., 1., 1., 1., 0.],
226         ...    [0., 0., 0., 0., 0., 0., 0.],
227         ... ]]])  # 1x1x7x7
228         >>> # compute the response map
229         >>> hessian_response(input)
Differences (unified diff with -expected +actual):
    @@ -1,7 +1,14 @@
    -tensor([[[[0.0155, 0.0334, 0.0194, 0.0000, 0.0194, 0.0334, 0.0155],
    -          [0.0334, 0.0575, 0.0339, 0.0000, 0.0339, 0.0575, 0.0334],
    -          [0.0194, 0.0339, 0.0497, 0.0000, 0.0497, 0.0339, 0.0194],
    -          [0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000, 0.0000],
    -          [0.0194, 0.0339, 0.0497, 0.0000, 0.0497, 0.0339, 0.0194],
    -          [0.0334, 0.0575, 0.0339, 0.0000, 0.0339, 0.0575, 0.0334],
    -          [0.0155, 0.0334, 0.0194, 0.0000, 0.0194, 0.0334, 0.0155]]]])
    +tensor([[[[-5.6396e-02, -7.5928e-02, -2.5255e-02, -0.0000e+00, -2.5255e-02,
    +           -7.5928e-02, -5.6396e-02],
    +          [-7.5928e-02, -3.2959e-02,  3.3339e-02,  0.0000e+00,  3.3339e-02,
    +           -3.2959e-02, -7.5928e-02],
    +          [-2.5255e-02,  3.3339e-02,  5.4160e-02,  0.0000e+00,  5.4160e-02,
    +            3.3339e-02, -2.5255e-02],
    +          [-3.1225e-17, -3.1225e-17, -3.1225e-17,  0.0000e+00, -3.1225e-17,
    +           -3.1225e-17, -3.1225e-17],
    +          [-2.5255e-02,  3.3339e-02,  5.4160e-02,  0.0000e+00,  5.4160e-02,
    +            3.3339e-02, -2.5255e-02],
    +          [-7.5928e-02, -3.2959e-02,  3.3339e-02,  0.0000e+00,  3.3339e-02,
    +           -3.2959e-02, -7.5928e-02],
    +          [-5.6396e-02, -7.5928e-02, -2.5255e-02, -0.0000e+00, -2.5255e-02,
    +           -7.5928e-02, -5.6396e-02]]]])

/git/kornia/kornia/feature/responses.py:229: DocTestFailure
```

The following one required plt which is not included in our package. Might need to be removed or updated.
```
________________________________________________________________________________________________________ [doctest] kornia.feature.laf.get_laf_pts_to_draw ________________________________________________________________________________________________________
283 
284     Returns:
285         pts: (torch.Tensor) tensor of boundary points
286 
287     Shape:
288         - Input: :math:`(B, N, 2, 3)`
289         - Output:  :math:`(B, N, n_pts, 2)`
290 
291     Examples:
292         >>> x, y = get_laf_pts_to_draw(LAF, img_idx)
UNEXPECTED EXCEPTION: NameError("name 'LAF' is not defined")
Traceback (most recent call last):

  File "/anaconda3/envs/torch/lib/python3.7/doctest.py", line 1329, in __run
    compileflags, 1), test.globs)

  File "<doctest kornia.feature.laf.get_laf_pts_to_draw[0]>", line 1, in <module>

NameError: name 'LAF' is not defined

/git/kornia/kornia/feature/laf.py:292: UnexpectedException
```

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

**Comment by edgarriba** [comment]
@shijianjian will take this and fix

**Comment by shijianjian** [comment]
Any update on this?

**Comment by edgarriba** [comment]
I will resume kornia development by the end of this week. Starting probably from this

**Comment by shijianjian** [comment]
@edgarriba The aforementioned failed tests have been taken out. Please check if they are giving the right numbers.

Besides, please try retrieving ```rtvec_to_pose``` function. Otherwise, the pinhole module will be broken.

**Comment by edgarriba** [comment]
@shijianjian please open an issue to track the missing functionalities and tag dmytro and me there and attach it to v0.5 release

**Review by edgarriba — APPROVED** [review]
LGTM
